apps/air-discount-scheme/backend/src/app/modules/discount/discount.model.ts

```diff
@@ -37,3 +50,61 @@ export class Discount implements TDiscount {
   @ApiProperty()
   expiresIn: number
 }
+
+interface ExplicitCodeCreationAttributes
+  extends Optional<TExplicitCode, 'id' | 'created' | 'modified' | 'flightId'> {}
+
+@Table({ tableName: 'explicit_code' })
+export class ExplicitCode
+  extends Model<TExplicitCode, ExplicitCodeCreationAttributes>
+  implements TExplicitCode {
+  @Column({
+    type: DataType.UUID,
+    primaryKey: true,
+    allowNull: false,
+    defaultValue: DataType.UUIDV4,
+  })
+  id!: string
+
+  @ForeignKey(() => Flight)
+  @Column({
+    type: DataType.UUID,
+    allowNull: true,
+  })
+  flightId?: string
+
+  @BelongsTo(() => Flight)
+  flight: any
```

Comment:
:sob:  `any`

Reply:
Yes, unfortunately :sob:

Reply:
Setting this to `flight: Flight` requires an initialiser or an assertion, but since this is connected later via an update such an assertion would be false.

That being said I don't remember if I tried
`flight?: Flight`.

There are other places in the codebase with `@BelongsTo(` that seem to have run into the same problem and are `any` today as well.

Reply:
Ok, nice catch.
Changed it accordingly.